### PR TITLE
Sandbox branch

### LINKED ISSUE #24: Design RotationalEncoders interface
Should look something like this:

```C++
class RotationalEncoder
{
  virtual Returns<void> Initialize() = 0;
  virtual Returns<units::degrees> GetRotation() = 0;
  // ... Add more here ...
};
```

### PR DESCRIPTION
**Pull request by joshua-git0**
Resolves SJSURoboticsTeam/urc-central-2021#24

### CODE REVIEW

**Comment by kammce** [comment]
@josh-github1 can you rename the title of this project. Also, you need to associate your commit with the issue number in central.

```
Resolves SJSURoboticsTeam/urc-central-2021/issues/#24
```

To your commit message or at least to your PR. so edit the PR description to include the message above.

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,35 @@
+BasedOnStyle: Google
```
Comment (kammce) [inline .clang-format:1]: Remove clang from this PR. You will need to rebase your branch from master as it already has clang format in it.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+!tools/launch_openocd_gdb.sh
+!tools/generate_doxygen_docs.sh
+!tools/doxygen_ignore_list.txt
+!tools/common.sh
+!tools/find_sources_without_tests.sh
+!tools/print_unused_test_files.sh
+!tools/gdb_dashboard
+!spell_checker.sh
+
+api/doxygen_stderr.txt
```
Comment (kammce) [inline .gitignore:34]: you can remove from your PR. Add it as a new PR.

File: Common/RotationalEncoder.hpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+class RotationalEncoder
+{
+ public:
+  virtual Returns<void> Initialize()            = 0;
+  virtual Returns<units::degrees> GetRotation() = 0;
+
+  virtual Returns<void> Enable() = 0;
+  virtual void SetAngleToZero()  = 0;
+}
+}  // namespace sjsu
```
Comment (kammce) [inline Common/RotationalEncoder.hpp:19]: Add a new line at the end of this file.

File: Common/RotationalEncoder.hpp [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+#include "utility/status.hpp"
+
+namespace sjsu
+{
+class RotationalEncoder
+{
+ public:
+  virtual Returns<void> Initialize()            = 0;
+  virtual Returns<units::degrees> GetRotation() = 0;
```
Comment (kammce) [inline Common/RotationalEncoder.hpp:14]: This line will not compile. Test this out in SJSU-Dev2 if you need to but you will see that it does not work.

File: Common/RotationalEncoder.hpp [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <cmath>
+#include <cstdint>
+
+#include "utility/status.hpp"
+
```
Comment (kammce) [inline Common/RotationalEncoder.hpp:7]: Your are going to need the "utility/units.hpp" library included here as well.

File: Common/RotationalEncoder.hpp [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <cmath>
+#include <cstdint>
```
Comment (kammce) [inline Common/RotationalEncoder.hpp:4]: I don't see you using cmath or cstdint in this file. Maybe you can remove these.

File: README.md [hunk 7]
```diff
@@ -0,0 +1,3 @@
+# SJSU Robotics Control Systems 2021
+
+Add more content later.
```
Comment (kammce) [inline README.md]: Not sure why this file is in here. You may need to pull the latest master branch, then rebase off of that.

**Review by kammce — CHANGES_REQUESTED** [review]

File: .clang-format (hunk 1 above)
Comment (joshua-git0) [reply .clang-format:1]: I have a lot to learn.. I just looked up git rebasing

Comment (kammce) [reply .clang-format:1]: This is still in here.

File: .gitignore (hunk 2 above)
Comment (kammce) [reply .gitignore:34]: This is still in here.

**Review by kammce — CHANGES_REQUESTED** [review]
You haven't resolved any of my comments.

**Comment by naterpotatoers** [comment]
going to close this PR since I don't think it will ever get merged

**Pull request closed without merging**